packages/dd-trace/src/plugins/util/web.js

```diff
@@ -432,6 +434,15 @@ function addRequestTags (context) {
     [HTTP_USERAGENT]: req.headers['user-agent']
   })
 
+  // if client ip has already been set by appsec, no need to run it again
+  if (config.clientIpEnabled && !span.context()._tags.hasOwnProperty(HTTP_CLIENT_IP)) {
```

Comment:
I think this should be safe to simplify:

```js
if (config.clientIpEnabled && !span.context()._tags[HTTP_CLIENT_IP]) {
```

That's the usual pattern for accessing `._tags` based on a grep search.

Reply:
I beg to disagree, I've written it with hasOwnProperty for a very specific purpose:
In some cases, the IP cannot be determined, so we set the tag as `undefined`. It will get stripped by the serializer before being sent to the backend, so it is absent from the trace, but it is still present in the `_tags` object.
See in [src/appsec/index.jsL81](https://github.com/DataDog/dd-trace-js/blob/e9a3413b8a7aed61e9d663e654089f639acade74/packages/dd-trace/src/appsec/index.js#L81):
![image](https://user-images.githubusercontent.com/70517128/214929229-1ee6bca9-c2ba-4d15-9cfe-f62cb7553872.png)
(`extractIp()`  can return `undefined` when it cannot determine the client IP)

When that happens, there is no need to try to extract the IP again in `web.js`, as we know for sure the result will be undefined again. Hence I'm using `hasOwnProperty()` to check if the property is set or not, regardless of the actual value.